sdk/sdk-core/aar/src/main/java/mil/emp3/core/mapgridlines/AbstractMapGridLine.java

```diff
@@ -0,0 +1,376 @@
+import mil.emp3.mapengine.interfaces.ICoreMapGridLineGenerator;
+import mil.emp3.mapengine.interfaces.IMapGridLines;
+import mil.emp3.mapengine.interfaces.IMapInstance;
+
+/**
+ * This abstract class is the base class for all map grid line generator classes.
+ */
+
+public abstract class AbstractMapGridLine implements IMapGridLines, ICoreMapGridLineGenerator {
+    private static final String TAG = AbstractMapGridLine.class.getSimpleName();
+
+    // The pixel density of the device.
+    protected static final int PIXELS_PER_INCH = Resources.getSystem().getDisplayMetrics().densityDpi;
+
+    // Object type for abstract components.
+    private static final String MAIN_GRID_TYPE_LABEL = "MAIN.gridtype.label";
+
+    // The map instance where the map grid is displayed.
+    protected final IMapInstance mapInstance;
+
+    private static final Handler uiLooper = new Handler(Looper.getMainLooper());
+
+    // The most recent time the grid was updated.
+    private Date lastUpdated;
+    // The list of features that make up the grid.
+    private final List<IFeature> featureList = new ArrayList<>();
+    // A list that is used to return the feature list to the map.
+    private final List<IFeature> returnList = new ArrayList<>();
+
+    // The camera setting when the view change occurred.
+    protected final ICamera currentCamera;
+    // The camera setting of the previous view change.
+    protected final ICamera previousCamera;
+    // The bounding box of the map.
+    private final EmpBoundingBox boundingBox;
+    // The meters per pixel across the center.
+    private double metersPerPixel;
+    // The generation thread.
+    private GridLineGenerationThread generationThread;
+
+    // Map list that hold the stroke style used by the grid lines.
+    private final Map<String, IGeoStrokeStyle> strokeStyleMap;
+    // Map list that hold the label styles used by the grid labels.
+    private final Map<String, IGeoLabelStyle> labelStyleMap;
+
+    private class GridLineGenerationThread extends java.lang.Thread {
+        private final Semaphore processEvent;
+        private boolean NotDone = true;
+
+        protected GridLineGenerationThread() {
+            this.setName("GridLineGenerationThread");
+            this.processEvent = new Semaphore(0);
+        }
+
+        @Override
+        public void run() {
+            while (NotDone) {
+                try {
+                    this.processEvent.acquire();
+
+                    long startTS = System.currentTimeMillis();
+                    String temp = String.format("North: %1$10.8f South: %2$10.8f West: %3$10.8f East: %4$10.8f", boundingBox.getNorth(), boundingBox.getSouth(), boundingBox.getWest(), boundingBox.getEast());
+                    Log.i(TAG, temp);
+                    processViewChange(boundingBox, currentCamera, metersPerPixel);
+                    Log.i(TAG, "feature generation in " + (System.currentTimeMillis() - startTS) + " ms. " + featureList.size() + " features.");
```

Comment:
Need to make a copy of boundingBox and currentCamera else mapViewChange may over-write this while it is being processed.